packages/argo-checkout-react/README.md

```diff
@@ -0,0 +1,92 @@
+
+`@shopify/argo-checkout-react` provides a few additional utilities if you are writing an Argo extension in React (it also re-exports all the relevant values from [`@shopify/argo-checkout`](../argo-checkout)). Before you start down this path, though, make sure you are working on an extension that will be complex enough to warrant the additional bundle size of including React. The [core API of remote-ui](https://github.com/Shopify/remote-ui/tree/main/packages/core) is built to be usable directly, and remote-ui provides other bindings, like [this one to `htm`](https://github.com/Shopify/remote-ui/tree/main/packages/htm), that provide some of the ergonomics of React without so much overhead. For additional details, please read the [performance guide](../argo-checkout/documentation/performance.md).
+
+To use the React bindings in Argo for Checkout, start by importing `React` as you normally would. All of the [core features of React](https://reactjs.org/docs/getting-started.html) are available, including hooks, context, and more.
+
+You’ll then import `render` from `@shopify/argo-checkout-react`. This function is a [thing wrapper](./src/render.ts) around [`shopify.extend`](../argo-checkout/documentation/globals.md) and [`@remote-ui/react`’s `render()`](https://github.com/Shopify/remote-ui/tree/main/packages/react#render). You’ll pass this function the name of an extension that can render UI, and a function that should return the JSX to render when that extension point is run. This function receives the input argument for the extension point.
+
+```tsx
+import {render} from '@shopify/argo-checkout';
+
+// `extensionPoint` is part of the [standard API](../argo-checkout/src/extension-points/api/standard)
+render('Checkout::PostPurchase::Render', ({extensionPoint}) => (
+  <App extensionPoint={extensionPoint} />
+));
+
+interface Props {
+  extensionPoint: string;
+}
+
+function App({extensionPoint}: Props) {
+  return <>Extension point: {extensionPoint}</>;
+}
+```
+
+If you’ve ever used React on the web, you’re probably used to returning DOM nodes as part of your React components. Because Argo extensions execute in a web worker and have no access to the DOM, returning DOM components is an error in Argo extensions. Instead, you can return the components you import from `@shopify/argo-checkout-react`, which are the equivalent of the DOM in Argo — they are the “leaf” elements, the lowest-level UI primitives that exist.
+
+```tsx
+import {render, Button} from '@shopify/argo-checkout';
+
+render('Checkout::PostPurchase::Render', (props) => <App {...props} />);
+
+interface Props {
+  extensionPoint: string;
+}
+
+function App({extensionPoint}: Props) {
+  return (
+    <Button
+      onPress={() => {
+        console.log(`Extension point: ${extensionPoint}`);
+      }}
+    >
+      Log extension point to console
+    </Button>
+  );
+}
+```
+
+## Other React-specific APIs
+
+### `useExtensionInput()`
```

Comment:
We need to rename it with `useExtensionApi` in this section :)

Reply:
Great catch!